internal/internal_task_handlers.go

```diff
@@ -1822,9 +1822,11 @@ func (wth *workflowTaskHandlerImpl) completeWorkflow(
 ) interface{} {
 	// for query task
 	if task.Query != nil {
+		cause := errToWorkflowTaskFailedCause(workflowContext.err)
```

Comment:
Nit: To test you might want to use a gRPC interceptor to verify the cause is sent since it won't be written to history